modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/java/com/liferay/frontend/js/svg4everybody/web/internal/servlet/taglib/SVG4EverybodyTopHeadDynamicInclude.java

```diff
@@ -78,7 +88,14 @@ public void include(
 			}
 		}
 
-		if (cdnHostEnabled) {
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		ThemeSpritemapCET themeSpritemapCET = _getThemeSpritemapCET(
+			themeDisplay.getLayout());
+
+		if (cdnHostEnabled || themeSpritemapCET.getEnableSVG4Everybody()) {
```

Comment:
I was wondering if we shall remove the responsibility of saying "this theme SVG client extension needs svg4everybody" from the client extension developer, yet turning this into a per-instance admin flag managed by sys admins/deployers instead.

In the end, if the need of svg4everybody is not known in advance, it's very likely devs will set that to true by default, however,  when working on the client extension, dev have little/no knowledge about how the system is going to deploy/serve it, nor the existence of a CDN, etc.

Makes sense?

Reply:
Is adding svg4everybody something innocuous? I mean, if you add it and it is not used, does it break anything? 

Because if that is the case we can move it to System Settings like we do with ESM shims, for instance...

Reply:
> dev have little/no knowledge about how the system is going to deploy/serve it, nor the existence of a CDN, etc.

Is this true? I would expect that whoever is adding the CET in portal and configuring it to be enabled on the page would also know if the URL they are adding is same domain or hosted somewhere entirely different.

I'm not opposed to moving it to a system setting, but as an end user I suspect it is annoying to add the CET and then also having to configure another setting somewhere else

Reply:
Well, the CE is added using a workspace and the dev probably has no idea of how everything is deployed/wired together and/or that wiring may change after the dev has finished the job. So yes, it could be false, but it will usually be true, I think.

> but as an end user I suspect it is annoying to add the CET and then also having to configure another setting somewhere else

More than annoying it can be a source of problems because it far from evident that the CE will fail unless you tweak some arcane setting in System Settings. Not sure how could we mitigate that (a warning somewhere? a proper error log explaining how to fix it? ...)

That said, once you turn that setting on you can forget about it and that's an advantage, over having to say if you want to use `svg4everybody` in every CE...

Is adding `svg4everybody` by default so bad? :thinking: Bc maybe we can just enable it by default...